memote/suite/tests/test_matrix.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2017 Novo Nordisk Foundation Center for Biosustainability,
+# Technical University of Denmark.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for matrix properties performed on an instance of ``cobra.Model``."""
+
+from __future__ import absolute_import, division
+
+import memote.support.matrix as matrix
+from memote.utils import annotate, wrapper
+
+
+@annotate(title="Ratio between largest and smallest non-zero coefficients",
+          type="percent"
+          )
+def test_absolute_extreme_coefficient_ratio(model):
+    """Show ratio of the absolute largest and smallest non-zero coefficients.
+
+    This test will return the absolute largest and smallest, non-zero
+    coefficients from the S-Matrix. A large ratio of these values may point to
+    potential numerical issues when trying to solve the underlying system of
+    equations.
+
+    To pass this test the ratio should not exceed 10^9. This threshold has
+    been selected based on experience, and is likely to be adapted when more
+    data on solver performance becomes available.
+    """
+    ann = test_absolute_extreme_coefficient_ratio.annotation
+    ann["data"] = matrix.absolute_extreme_coefficient_ratio(model)
+    ann["metric"] = ann["data"][0] / ann["data"][1]
+    ann["message"] = wrapper.fill(
+        """The ratio of the absolute highest coefficient {} and the lowest,
+        non-zero coefficient {} is: {}""".format(
+            ann["data"][0], ann["data"][1], ann['metric']
+        ))
+    assert ann["metric"] < 1e9, ann["message"]
+
+
+@annotate(title="Number of independent conservation relations in model",
+          type="number"
+          )
+def test_number_independent_conservation_relations(model):
+    """Show number of independent conservation relations in the model.
+
+    This test will return the number of conservation relations, i.e.
+    conservation pools through the left null space of the S-Matrix.
+
+    This test is not scored, as the dimension of the left null space
+    depends on the S-Matrix constructed, which is system-specific.
+    """
+    ann = test_number_independent_conservation_relations.annotation
+    ann["data"] = matrix.number_independent_conservation_relations(model)
+    ann["metric"] = ann["data"].shape[1]
```

Comment:
I think this operation should probably take part in the helper function itself:
```ann["data"].shape[1]```

In other words I think a test named `number_independent_conservation_relations` should probably return an integer and not a matrix. If we need the matrix somewhere else we should probably split up the helper functions into one that generates the left nullspace matrix and one that extracts the corresponding number of relations from it?

Moreover, `ann["metric"]` should only be assigned if there is a ratio/ percentage worth showing or the test itself is scored. In this case, I'd leave it unassigned. Take a look at one of my comments below where I quote the docstring from the `annotate` function in `utils.py`. Perhaps that explains a bit better what `ann["data"]` can and cannot be.